lib/percy/client/environment.rb

```diff
@@ -129,8 +132,8 @@ def self.branch
         end
 
         if result == ''
-          STDERR.puts '[percy] Warning: not in a git repo, setting PERCY_BRANCH to "master".'
-          result = 'master'
+          STDERR.puts '[percy] Warning: not in a git repo, no BRANCH detected.'
+          result = BRANCH_FALLBACK
```

Comment:
Would recommend simplifying this a bit by: dropping BRANCH_FALLBACK, just doing `return` here, and expect it to `be_nil` in the test. No real need to have a constant for it if we just are introducing the ability for it to be nil.